lib/services/nodes-api-service.js

```diff
@@ -30,64 +32,161 @@ function nodeApiServiceFactory(
     }
 
     /**
-     * Find Enclosure nodes which enclose compute node
+     * Find nodes downstream that relate with the removing node
      * @param  {Object}     node
-     * @return {Promise}
+     * @param  {Object}     type
+     * @return {Promise}    array of downstream nodes
      */
-    NodeApiService.prototype._findEnclNodes = function(node) {
+    NodeApiService.prototype._findDsNodes = function(node, type) {
```

Comment:
I have thought the name a lot before when coding, and got to know from HW engineer that "downstream port" is abbreviated as "DSP", thus I chose the variable name here as it is now (ds). But now it seems that it still brings confusion, changing the function name to "downstream" for clear understanding and keeping the local variable as "ds" for direct use might be good choice.